DUnitX.Assert.pas

```diff
@@ -48,21 +49,22 @@ interface
   Assert = class
   private
     class var fOnAssert: TProc;
-    class var fTestFailure: ExceptClass;
```

Comment:
If Assert is going to need to stay framework-agnostic, then yes my changes will have to be done differently.  I can't really go the helper route because you can only have one helper per class and the helper would also have to be included in the project or a uses clause (could confuse people), and that would limit anyone else from adding a helper later.  One of my goals was to integrate this directly in Assert.  While you consider the diffing logic a special case, it is not only for diffing, that is just how the GUI logger uses it.  It could be used by automated systems to save the data out and add as artifacts in the CI so people could review the output in a more digestible format. The problem is the data that you are asserting is locked in a string error message which makes it hard to act on if desired.

While I knew Assert was supposed to remain as self-contained as possible (which it still is), I did not know it was supposed to remain framework independent (which it now is not because of the exceptions).  I will see about getting the latter back - I am not exactly sure the best way to do that but will explore some options.